File: app/src/main/java/me/imzack/app/ender/injector/component/TypeDetailComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.TypeDetailPresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.TypeDetailActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = TypeDetailPresenterModule.class)
public interface TypeDetailComponent {

    void inject(TypeDetailActivity typeDetailActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/TypeDetailComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.TypeDetailPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.TypeDetailActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TypeDetailPresenterModule::class))
interface TypeDetailComponent {

    fun inject(typeDetailActivity: TypeDetailActivity)
}
